### DISCUSSION TITLE
Save my state in session

### DISCUSSION THREAD

**Discussion by ghost** [orbitalquark/textadept#340] [Q&A] [answered]
Hi.
In my personal init.lua i have:

`local currentTag = 'div'

keys['ctrl+1'] = function()
  currentTag = ui.dialogs.input{title = 'Set Current Tag:', text = currentTag}
end`

Any way to persist `currentTag` in session?

Assistance greatly appreciated.

**Comment by orbitalquark** [comment, marked as answer]
You can connect to `events.SESSION_SAVE` and `events.SESSION_LOAD` to 
persist data in a session. 
(https://orbitalquark.github.io/textadept/api.html#textadept.session) 
Something like this may work:

     events.connect(events.SESSION_SAVE, function(session)
       session.currentTag = currentTag
     end)
     events.connect(events.SESSION_LOAD, function(session)
       currentTag = session.currentTag
     end)

**Reply by ghost** [reply]
Thanks much.
Reading the manual.

**Reply by ghost** [reply]
It's really frustrating to be so unproductive
in a programming language. I'm two weeks into
my Lua career.

Thank you for a wonderful product, and the
opportunity to learn Lua: something I've been
wanting to do for some time.

Keeping PIL 4ed., Lua5.4 Reference Manual,
Textadept API open at all times. Going to
be a long haul.

**Reply by ghost** [reply]
Successfully installed textadept-debugger and
stepped through my first-debug-session.lua!

Path forward looks rosy, indeed! :)

**Comment by merlindiavova** [comment]
Did this work for you?

I cannot seem to get the saving part to work?

**Reply by merlindiavova** [reply]
I did get this to work for me finally.
Realized I was using `return` in the in the events.SESSION_SAVE in another module.
Removing that, this works perfectly.

Lesson learned do not return when connecting to event unless you you want to prevent other events!
